### ISSUE TITLE
Ray Data v 2.9.1 conflict with pandas 2.2

### ISSUE THREAD

**Issue by jq** [ray-project/ray#42956]
### What happened + What you expected to happen

Just install the  `pip install -U "ray[data,train,tune,serve]"` the latest version and run the ray data sample 
```
# Create datasets from on-disk files, Python objects, and cloud storage like S3.
ds = ray.data.read_csv("s3://anonymous@ray-example-data/iris.csv")

# Apply functions to transform data. Ray Data executes transformations in parallel.
def compute_area(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
    length = batch["petal length (cm)"]
    width = batch["petal width (cm)"]
    batch["petal area (cm^2)"] = length * width
    return batch

transformed_ds = ds.map_batches(compute_area)

# Iterate over batches of data.
for batch in transformed_ds.iter_batches(batch_size=4):
    print(batch)

# Save dataset contents to on-disk files or cloud storage.
transformed_ds.write_parquet("local:///tmp/iris/")
```
hit error
```
2024-02-02 15:43:12,042	INFO worker.py:1715 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265 
2024-02-02 15:43:19,316	INFO set_read_parallelism.py:115 -- Using autodetected parallelism=32 for stage ReadCSV to satisfy parallelism at least twice the available number of CPUs (16).
2024-02-02 15:43:19,316	INFO set_read_parallelism.py:122 -- To satisfy the requested parallelism of 32, each read task output is split into 32 smaller blocks.
2024-02-02 15:43:19,317	INFO streaming_executor.py:112 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[ReadCSV] -> TaskPoolMapOperator[MapBatches(compute_area)]
2024-02-02 15:43:19,318	INFO streaming_executor.py:113 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=None, gpu=None, object_store_memory=None), exclude_resources=ExecutionResources(cpu=0, gpu=0, object_store_memory=0), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=False)
2024-02-02 15:43:19,318	INFO streaming_executor.py:115 -- Tip: For detailed progress reporting, run `ray.data.DataContext.get_current().execution_options.verbose_progress = True`
[dataset]: Run `pip install tqdm` to enable progress reporting.
2024-02-02 15:43:19,642	ERROR streaming_executor_state.py:476 -- An exception was raised from a task of operator "ReadCSV->SplitBlocks(32)". Dataset execution will now abort. To ignore this exception and continue, set DataContext.max_errored_blocks.
Traceback (most recent call last):
  File "/Users/jqian2/Library/CloudStorage/GoogleDrive-jqian2@snapchat.com/My Drive/work/example-code-2e/0-exp/ray_data.py", line 18, in <module>
    for batch in transformed_ds.iter_batches(batch_size=4):
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/iterator.py", line 164, in _create_iterator
    block_iterator, stats, blocks_owned_by_consumer = self._to_block_iterator()
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/iterator/iterator_impl.py", line 33, in _to_block_iterator
    block_iterator, stats, executor = ds._plan.execute_to_iterator()
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/plan.py", line 559, in execute_to_iterator
    block_iter = itertools.chain([next(gen)], gen)
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/legacy_compat.py", line 61, in execute_to_legacy_block_iterator
    for bundle in bundle_iter:
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/interfaces/executor.py", line 37, in __next__
    return self.get_next()
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/streaming_executor.py", line 145, in get_next
    item = self._outer._output_node.get_output_blocking(
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/streaming_executor_state.py", line 287, in get_output_blocking
    raise self._exception
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/streaming_executor.py", line 212, in run
    while self._scheduling_loop_step(self._topology) and not self._shutdown:
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/streaming_executor.py", line 260, in _scheduling_loop_step
    num_errored_blocks = process_completed_tasks(
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/streaming_executor_state.py", line 477, in process_completed_tasks
    raise e from None
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/streaming_executor_state.py", line 444, in process_completed_tasks
    num_blocks_read = task.on_data_ready(
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/interfaces/physical_operator.py", line 102, in on_data_ready
    raise ex from None
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/interfaces/physical_operator.py", line 98, in on_data_ready
    ray.get(block_ref)
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/_private/auto_init_hook.py", line 22, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/_private/client_mode_hook.py", line 103, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/_private/worker.py", line 2624, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(ImportError): ray::ReadCSV->SplitBlocks(32)() (pid=10630, ip=127.0.0.1)
    for b_out in map_transformer.apply_transform(iter(blocks), ctx):
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/operators/map_transformer.py", line 430, in __call__
    for block in blocks:
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/execution/operators/map_transformer.py", line 377, in __call__
    yield output_buffer.next()
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/output_buffer.py", line 73, in next
    block_to_yield = self._buffer.build()
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/delegating_block_builder.py", line 64, in build
    return self._builder.build()
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/table_block.py", line 123, in build
    return self._concat_tables(tables)
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/arrow_block.py", line 148, in _concat_tables
    return transform_pyarrow.concat(tables)
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/_internal/arrow_ops/transform_pyarrow.py", line 155, in concat
    from ray.data.extensions import (
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/extensions/__init__.py", line 1, in <module>
    from ray.data.extensions.tensor_extension import (
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/data/extensions/tensor_extension.py", line 7, in <module>
    from ray.air.util.tensor_extensions.pandas import (  # noqa: F401
  File "/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/ray/air/util/tensor_extensions/pandas.py", line 45, in <module>
    from pandas.io.formats.format import ExtensionArrayFormatter
ImportError: cannot import name 'ExtensionArrayFormatter' from 'pandas.io.formats.format' (/usr/local/Caskroom/miniconda/base/envs/r/lib/python3.9/site-packages/pandas/io/formats/format.py)
```
I tried python 3.9 3.10 3.11 all same errors

### Versions / Dependencies

Ray 2.9.1
python 3. 9 3.10 3.11
OS: mac sonoma 14.3
pandas pandas-2.2.0

### Reproduction script

clean new python env. 
`pip install -U "ray[data,train,tune,serve]"`
```
# Create datasets from on-disk files, Python objects, and cloud storage like S3.
ds = ray.data.read_csv("s3://anonymous@ray-example-data/iris.csv")

# Apply functions to transform data. Ray Data executes transformations in parallel.
def compute_area(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
    length = batch["petal length (cm)"]
    width = batch["petal width (cm)"]
    batch["petal area (cm^2)"] = length * width
    return batch

transformed_ds = ds.map_batches(compute_area)

# Iterate over batches of data.
for batch in transformed_ds.iter_batches(batch_size=4):
    print(batch)

# Save dataset contents to on-disk files or cloud storage.
transformed_ds.write_parquet("local:///tmp/iris/")
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by amitschang** [comment]
FYI, this looks like a duplicate of https://github.com/ray-project/ray/issues/42842, which points to the merged fix

**Comment by scottjlee** [comment]
Dupe of https://github.com/ray-project/ray/issues/42842, which was fixed by https://github.com/ray-project/ray/pull/42892.
Please try merging in latest master, and feel free to re-surface the discussion in the above issue if you continue to run into the problem. Thanks!

**Comment by scottjlee** [comment]
Dupe of https://github.com/ray-project/ray/issues/42842, which was fixed by https://github.com/ray-project/ray/pull/42892.
Please try merging in latest master, and feel free to re-surface the discussion in the above issue if you continue to run into the problem. Thanks!
